### ISSUE TITLE
Failed to create a DefaultVideoTransformDevice: TypeError: Converting circular structure to JSON

### ISSUE THREAD

**Issue by tarak-bit** [aws/amazon-chime-sdk-component-library-react#756]
### What are you trying to do?
I am trying to implement the feature of background blur and background replacement referencing the demo meeting app

https://github.com/aws-samples/amazon-chime-sdk/tree/main/apps/meeting#readme---react-meeting-demo 

### How can the documentation be improved to help your use case?

The app i am trying to incorporate the feature is in javascript. I am facing the error **create a DefaultVideoTransformDevice: TypeError: Converting circular structure to JSON** while selecting the background blur or replacement option from the option dropdown.
![image](https://user-images.githubusercontent.com/96565976/158119380-87b04f45-e3d7-4ae6-9ef5-bd66c295f493.png)

After trying a lot i found one work around to solve the issue. In index.esm.js file when we pass the 2nd argument a replacer function in JSON.stringify it works. Please refer the below code (createBackgroundBlurDevice function) for clear understanding

    current code:  1.console.log("Calling createBackgroundBlurDevice with device: " + JSON.stringify(selectedDevice));
                          2. console.log("Created video transform device " + JSON.stringify(chosenVideoTransformDevice, null, 2));
                          
    Replaced with : 1.  console.log("Calling createBackgroundBlurDevice with device: " + JSON.stringify(selectedDevice,replacerFunc()));
                              2.console.log("Created video transform device " + JSON.stringify(chosenVideoTransformDevice, replacerFunc()));

replacer functuion  adding after base spacing function:

```
var replacerFunc = function() {
    const visited = new WeakSet();
    return (key, value) => {
      if (typeof value === "object" && value !== null) {
        if (visited.has(value)) {
          return;
        }
        visited.add(value);
      }
      return value;
    };
  };

```

Also same things changed in createBackgroundReplacementDevice function do the work for background replacement

### What documentation have you looked at so far?

Amazon Chime SDK React Components Library GitHub 
 React SDK meeting demo
 Amazon Chime SDK React Components Library GitHub issues

**Comment by tarak-bit** [comment]
Commenting out the console line also do the work 

![createblur](https://user-images.githubusercontent.com/96565976/158132233-d0fffd3a-a0a7-4a5f-a72f-7f9128180531.png)

![image](https://user-images.githubusercontent.com/96565976/158132899-006818bf-6800-428f-8f44-ffaf7dda6963.png)

**Comment by tarak-bit** [comment]
@michhyun1 i am using Version 96.0.4664.110 (Official Build) (64-bit) of chrome. i also checked with other browsers as well. Same error persist and after commenting out that mentioned line from the **createBackgroundBlurDevice** and  **createBackgroundReplacementDevice**, the error gone away. 

I implemented the mentioned functionality by referencing the demo app( https://github.com/aws-samples/amazon-chime-sdk/tree/main/apps/meeting#readme---react-meeting-demo) and used almost the same code as here(in Javascript).

**Comment by ltrung** [comment]
@tarak-bit Thanks for submitting the issue. Just quick question is your Javascript code ES5 or ES6?
I could not reproduce it on our side but I think we target ES6 when compiling our code from TS to JS where `Set` is introduced. JSON.stringify does not do anything for a Set object so we do not run into this issue. So just want to check on your JS version.

**Comment by tarak-bit** [comment]
@ltrung i am using ES6 version of javascript in my code.

**Comment by ltrung** [comment]
Interesting. Could you try to use this code in your application and see whether you have any error and give us the output?

Also do you use any bundling tool like babel?
```
class B {
  observers = new Set();
  addObserver(observer) {
    this.observers.add(observer);
  }
}
class A {
  b = undefined;
  constructor() {
    this.b = new B();
    this.b.addObserver(this);
  }
}

const a = new A();
console.log(JSON.stringify(a));
console.log(a.b.observers.entries());

const set = new Set();
console.log(set);
console.log(JSON.stringify(set));
```

**Comment by tarak-bit** [comment]
HI @ltrung , Thanks for your continuous support

The above code in my project not run properly and JSON.stringify() line creates the same type of error. I am using CRA (create react app) to create the project and bundling tool BABEL is also getting used for this.

Error: 

![image](https://user-images.githubusercontent.com/96565976/160064108-5c84fae0-24dd-4dad-bd9a-5ada5e3111f6.png)

**Comment by samchienGG** [comment]
I have the same problem:see_no_evil:

**Comment by ltrung** [comment]
#768 to remove the circular JSON.stringify statement has been merged to `main`. Could you try it out? @samchienGG @tarak-bit

**Comment by tarak-bit** [comment]
@ltrung Thanks for the confirmation.

I can see that update is only merged in github but not updated in NPM package **amazon-chime-sdk-component-library-react**. could you please help me by releasing the same update in NPM package also.

**Comment by michhyun1** [comment]
@tarak-bit This fix will be released as part of our next release.

**Comment by LeviSklaroff** [comment]
This fix has been released in version 3.0.0 please try it out and let us know if there's any further issues.
